modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -621,9 +625,27 @@ protected void doInitialise() throws InitialisationException {
     flowConstruct = new LazyValue<>(() -> (FlowConstruct) componentLocator.find(getRootContainerLocation()).orElse(null));
     messageProcessContext = createProcessingContext();
     if (shouldRunOnThisNode()) {
+      if (LOGGER.isDebugEnabled()) {
+        boolean isPrimaryPollingInstance = clusterService.isPrimaryPollingInstance();
+        if (primaryNodeOnly) {
+          LOGGER
+              .debug("Message source '{}' on flow '{}' is being initialised, it is running on the primary node and it must only run on this node.",
```

Comment:
Suggestions:
1. Spelling "initializing"
2. "Message source '{}' on flow '{}' running on the primary node is initializing.  Note that Message source '{}' must run on the primary node only."

Reply:
Taking both suggestions.